src/objectExplorer/objectExplorerUtils.ts

```diff
@@ -25,21 +26,28 @@ export class ObjectExplorerUtils {
     }
 
     public static getNodeUri(node: TreeNodeInfo): string {
-        if (node.nodeType === Constants.disconnectedServerLabel) {
-            return undefined;
+        const profile = <IConnectionProfile>node.connectionCredentials;
+        return ObjectExplorerUtils.getNodeUriFromProfile(profile);
+    }
+
+    public static getNodeUriFromProfile(profile: IConnectionProfile): string {
+        const uri = profile.server + '_' +
+            profile.database + '_' +
+            profile.user + '_' +
+            profile.profileName;
+        return uri;
+    }
+
+    public static getDatabaseName(node: TreeNodeInfo): string {
+        if (node.nodeType === Constants.serverLabel) {
+            return node.connectionCredentials.database;
         }
         while (node) {
-            if (node.nodeType === Constants.serverLabel) {
-                break;
+            if (node.nodeType === Constants.databaseString) {
+                return node.label;
             }
```

Comment:
Isn't this going to have the same issue as you just fixed with the system versioned dbs?

Reply:
Not really. This function sets the name to `<default>` is no database is found. We set it back to '' in the fix I sent out. Empty databases are handled as default in connections.

Reply:
I'm saying in that you return node.label here - but as we saw before the label isn't always the actual DB name. You should be using node.metadata.name.

And actually I filed a bug in another place I noticed this - [https://github.com/microsoft/vscode-mssql/issues/1426](https://github.com/microsoft/vscode-mssql/issues/1426) - which brings up the question why we have two separate functions doing the same thing. 

So my suggestion is for you to just leave this as-is for now and then do a follow-up PR to consolidate those two into a single method and fix the issue there. 

---
In reply to: [334679533](https://github.com/microsoft/vscode-mssql/pull/1423#discussion_r334679533) [](ancestors = 334679533)

Reply:
For databases, the node label is always the database name. The issue arose only for objects such as tables, sprocs, view etc. But I'll consolidate the logic anyway.

Reply:
Thank you. The reason why we'd want to do this is that if we ever wanted to add something to the label in the future then this would just start silently start breaking. Using the metadata name here is definitely the correct approach.